istioctl/cmd/kubeinject_test.go

```diff
@@ -48,6 +48,29 @@ func TestKubeInject(t *testing.T) {
 				" "),
 			goldenFilename: "testdata/deployment/hello.yaml.injected",
 		},
+		{ // case 4 with only iop files
+			args: strings.Split(
+				"kube-inject --operatorFileName testdata/istio-operator.yaml"+
+					" --injectConfigFile testdata/inject-config-iop.yaml -f testdata/deployment/hello.yaml",
+				" "),
+			goldenFilename: "testdata/deployment/hello.yaml.iop.injected",
+		},
+		{ // case 5 with only iop files
+			args: strings.Split(
+				"kube-inject --operatorFileName testdata/istio-operator.yaml"+
+					" --injectConfigFile testdata/inject-config-inline-iop.yaml -f testdata/deployment/hello.yaml",
+				" "),
+			goldenFilename: "testdata/deployment/hello.yaml.iop.injected",
+		},
+		{ // case 6 with iops and values override
+			args: strings.Split(
+				"kube-inject --operatorFileName testdata/istio-operator.yaml"+
+					" --injectConfigFile testdata/inject-config-inline-iop.yaml"+
```

Comment:
seems like `injectConfigFile` shouldn't be *required* with operatorFileName? Maybe optional, but it can be derived?

Reply:
Yes, `injectConfigFile` is not required. If it is not provided, either `ExternalInjector` will be used or the injection template will be derived from `istio-sidecar-injector` config map. Providing `injectConfigFile` flag will help make the tests' want results become consistent because the templates are always the same.